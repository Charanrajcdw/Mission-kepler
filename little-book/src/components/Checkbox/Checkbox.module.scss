@import "../../scss/fonts";

.checkBoxGroup {
  font-family: $font-roboto;
  display: flex;
  align-items: center;

  input[type="checkbox"] {
    accent-color: var(--pink);
    margin: 7px 14px 7px 0;
    height: 20px;
    width: 20px;
  }

  label {
    font-size: 0.9rem;
    &::first-letter {
      text-transform: capitalize;
    }
  }
}
