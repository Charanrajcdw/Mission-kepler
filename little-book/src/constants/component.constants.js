export const MODAL = {
  members: "Members",
  newBlog: "Add New Blog",
  remove: "Remove",
};

export const MEMBERS_LIST = {
  title: "Members",
  noMembers: "No members available",
};

export const BLOG_FORM = {
  title: "Add New Blog",
  button: "ADD",
  titlePlaceholder: "Name your blog",
  detailsPlaceholder: "Write Content Here ..",
  invalidValues: "Invalid title or description",
  invalidBlog: "Blog title already exists",
  blogAdded: "New Blog added successfully",
  localBlog: "local",
};

export const WARNING_MODAL = {
  title: "Unsaved changes will be lost. Are you sure to continue?",
  continue: "Continue",
  cancel: "Cancel",
};
