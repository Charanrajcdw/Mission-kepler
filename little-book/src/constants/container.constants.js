export const SIDEBAR = {
  logo: "Little Book",
  filter: "FILTER",
  noFilter: "Add blogs to show filters",
  viewMembers: "View Members",
  switchToDark: "Switch to Dark Mode",
  switchToLight: "Switch to Light Mode",
  blogs: "Blogs",
};
