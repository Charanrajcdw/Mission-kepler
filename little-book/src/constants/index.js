export * from "./url.constants";
export * from "./theme.constants";
