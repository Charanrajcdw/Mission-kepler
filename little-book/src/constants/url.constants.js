export const API_URLS = {
  blogs: "https://jsonmockserver.vercel.app/api/blogs",
  users: "https://jsonmockserver.vercel.app/api/users",
  images: "https://jsonmockserver.vercel.app/",
};
