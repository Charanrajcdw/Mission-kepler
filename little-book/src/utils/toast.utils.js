import { toast } from "react-toastify";

export const showSuccessToast = (message, currentTheme) => {
  toast.success(message, {
    position: "top-right",
    autoClose: 2000,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    theme: currentTheme,
  });
};
