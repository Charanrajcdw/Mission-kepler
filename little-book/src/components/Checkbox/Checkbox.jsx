import { memo, useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import PropTypes from "prop-types";
import styles from "./Checkbox.module.scss";
import { SIDEBAR } from "../../constants";
import { blogActions } from "../../store";

const Checkbox = ({ label, modalHandler, isSelected }) => {
  const [isChecked, setIsChecked] = useState(isSelected);
  const isEditing = useSelector((state) => state.blogs.isEditing);
  const dispatch = useDispatch();

  useEffect(() => {
    setIsChecked(isSelected);
  }, [isSelected]);

  const changeHandler = () => {
    if (isEditing) {
      modalHandler(false, blogActions.modifyTypes(label));
      return;
    }
    setIsChecked((prevValue) => !prevValue);
    dispatch(blogActions.modifyTypes(label));
  };

  return (
    <div className={styles.checkBoxGroup}>
      <input type="checkbox" checked={isChecked} id={label} value={label} onChange={changeHandler} />
      <label htmlFor={label}>
        {label} {SIDEBAR.blogs}
      </label>
    </div>
  );
};

Checkbox.propTypes = {
  label: PropTypes.string.isRequired,
  modalHandler: PropTypes.func.isRequired,
  isSelected: PropTypes.bool.isRequired,
};

export const memoizedCheckbox = memo(Checkbox);

export default memoizedCheckbox;
