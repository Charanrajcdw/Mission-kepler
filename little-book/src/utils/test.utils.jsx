import { configureStore } from "@reduxjs/toolkit";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import ThemeContextProvider from "../contexts/theme.context";
import blogSlice from "../store/slices/blogSlice";
import { THEME } from "../constants";

const customRender = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
    ...options
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const customCreateSnapshot = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return renderer.create(<ProviderWrapper>{ui}</ProviderWrapper>).toJSON();
};

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render, customCreateSnapshot as createSnapshot };
