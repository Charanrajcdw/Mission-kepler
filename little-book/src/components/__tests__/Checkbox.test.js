import { render, screen, createSnapshot, userEvent } from "../../utils/test.utils.jsx";
import Checkbox from "../Checkbox/Checkbox";
import { SIDEBAR, WARNING_MODAL } from "../../constants";

describe("Checkbox", () => {
  it("should render Checkbox", () => {
    render(<Checkbox label="local" modalHandler={() => {}} />);
    const checkboxGroup = screen.getByLabelText(`local ${SIDEBAR.blogs}`);
    expect(checkboxGroup).toBeInTheDocument();
  });

  it("should match checkbox snapshot", () => {
    const checkbox = createSnapshot(<Checkbox label="local" modalHandler={() => {}} />);
    expect(checkbox).toMatchSnapshot();
  });

  it("should change checked state", async () => {
    const user = userEvent.setup();
    render(<Checkbox label="local" modalHandler={() => {}} />);
    const checkboxInput = screen.getByRole("checkbox");
    await user.click(checkboxInput);
    expect(checkboxInput).not.toBeChecked();
    await user.click(checkboxInput);
    expect(checkboxInput).toBeChecked();
  });

  it("should show warning modal if editing is true", async () => {
    const user = userEvent.setup();
    const initialBlogsData = { isEditing: true };
    render(<Checkbox label="local" modalHandler={() => {}} />, { preloadedState: { blogs: initialBlogsData } });
    const checkboxInput = screen.getByRole("checkbox");
    await user.click(checkboxInput);
    expect(checkboxInput).toBeChecked();
  });
});
