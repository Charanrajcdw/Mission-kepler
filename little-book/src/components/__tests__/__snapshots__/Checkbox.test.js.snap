// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match checkbox snapshot 1`] = `
<div
  className="checkBoxGroup"
>
  <input
    checked={true}
    id="local"
    onChange={[Function]}
    type="checkbox"
    value="local"
  />
  <label
    htmlFor="local"
  >
    local
     
    Blogs
  </label>
</div>
`;
